src/plugins/unified_histogram/public/container/container.tsx

```diff
@@ -141,6 +144,37 @@ export const UnifiedHistogramContainer = forwardRef<
     return null;
   }
 
+  if (
```

Comment:
@ppisljar this creates a bug when you try to edit the histogram (no stats mode). The bug will be solved if you change this to:

```
  if (
    currentSuggestion &&
    currentSuggestion.datasourceId === 'textBased' &&
    currentSuggestion.datasourceState &&
    (currentSuggestion.datasourceState as TextBasedPersistedState).layers &&
    containerProps.table &&
    Object.values((currentSuggestion.datasourceState as TextBasedPersistedState).layers).some(
      (l) => l.table !== containerProps.table
    )
  ) {
    const layerIds = Object.keys(
      (currentSuggestion.datasourceState as TextBasedPersistedState).layers
    );

    const hasTables = layerIds.some((layerId) => {
      const layer = (currentSuggestion!.datasourceState as TextBasedPersistedState).layers[layerId];
      return 'table' in layer && layer.table;
    });

    if (hasTables) {
      const newLayers: TextBasedPersistedState['layers'] = {};
      layerIds.forEach((layerId) => {
        newLayers[layerId] = {
          ...(currentSuggestion!.datasourceState as TextBasedPersistedState).layers[layerId],
          table: containerProps.table,
        };
      });

      stateService?.setCurrentSuggestion({
        ...currentSuggestion,
        datasourceState: {
          ...(currentSuggestion.datasourceState as TextBasedPersistedState),
          layers: newLayers,
        },
      });
    }
  }
```

You need to assign the new table and to update the suggestion only when there are already tables. 
Make this change, I will test for a last time and approve. Everything else seems to work great 🎉

Reply:
Or maybe is better to store in the state this value https://github.com/elastic/kibana/blob/main/src/plugins/unified_histogram/public/layout/hooks/use_lens_suggestions.ts#L146 

and do not run the above code when isOnHistogramMode

```
if (
    currentSuggestion &&
    currentSuggestion.datasourceId === 'textBased' &&
    currentSuggestion.datasourceState &&
    (currentSuggestion.datasourceState as TextBasedPersistedState).layers &&
    containerProps.table &&
   !isOnHistogramMode &&
    Object.values((currentSuggestion.datasourceState as TextBasedPersistedState).layers).some(
      (l) => l.table !== containerProps.table
    )
  ) {
 // run code when there is not on histogram mode
}
```

I think this is cleaner. As far as I can see, the table might not exist in the stats mode too so the first solution is not the best.